jib-core/src/test/java/com/google/cloud/tools/jib/event/progress/AllocationTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.event.progress;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link Allocation}. */
+public class AllocationTest {
+
+  /** Error margin for checking equality of two doubles. */
+  private static final double DOUBLE_ERROR_MARGIN = 1e-10;
+
+  /**
+   * Checks that the {@code allocations} add up to a full {@code 1.0}.
+   *
+   * @param allocations the leaf allocations
+   */
+  private static void checkAllocationComplete(Allocation... allocations) {
```

Comment:
This function is only called once, so maybe it can just go straight in the test.